lib/drivers/insuletDriver.js

```diff
@@ -614,18 +644,47 @@ module.exports = function (config) {
     });
   };
 
+  // these aren't history records, so we have to find them separately
+  var findPumpAlarmRecords = function (recordlist) {
+    var result = [];
+    for (var i = 0; i < recordlist.length; ++i) {
+      // TODO: check with KJQ about log_index of -1???
+      // doesn't have a timestamp, so I think it's not a valid record
```

Comment:
I think if it doesn't have a timestamp, it must not be valid. I can't imagine what we would do with it.